### PR TITLE
#333 convert favourites to kotlin

### PR DESCRIPTION
**Pull request by vishnus1224**

### CODE REVIEW

**Comment by rock3r** [comment]
@vishnus1224 could you please address the Detekt warnings?

```
Ruleset: complexity
	LongParameterList - 6/5 - [handleFavoriteButtonClick] at java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutPresenter.kt:9:39
Ruleset: empty-blocks
Ruleset: exceptions
Ruleset: performance
Ruleset: potential-bugs
Ruleset: style
	VariableNaming - [TYPE_TALK] at java/net/squanchy/favorites/view/FavoritesAdapter.kt:22:5
	VariableNaming - [TYPE_HEADER] at java/net/squanchy/favorites/view/FavoritesAdapter.kt:23:5
	NewLineAtEndOfFile - [FavoritesSignedInEmptyLayoutPresenter.kt] at java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutPresenter.kt:21:1
	NewLineAtEndOfFile - [FavoritesSignedInEmptyLayoutView.kt] at java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutView.kt:10:1
	NewLineAtEndOfFile - [FavoritesListView.kt] at java/net/squanchy/favorites/view/FavoritesListView.kt:34:1
	NewLineAtEndOfFile - [FavoritesPageView.kt] at java/net/squanchy/favorites/FavoritesPageView.kt:121:1
	NewLineAtEndOfFile - [FavoritesComponent.kt] at java/net/squanchy/favorites/FavoritesComponent.kt:34:1
```

File: app/src/main/java/net/squanchy/favorites/FavoritesComponent.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/FavoritesComponent.kt:25]: no empty line needed here

File: app/src/main/java/net/squanchy/favorites/FavoritesComponent.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/FavoritesComponent.kt]: empty line

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt]: as a layout for classes in Kotlin we use:

```
class ClassName @JvmOverloads constructor(
    param1: Class1,
    ...,
    paramX: ClassX
) : SuperClass(param1, ..., paramX), Interface {   
```

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt]: there's no need to use `lazy` here, just do:
`private val service: ScheduleService = favoritesComponent.service()`

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt]: new line after `disposable.add(`

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt]:
```
 return when (menuItem.itemId) {
     R.id.action_search -> { showSearch(); true }
     R.id.action_settings -> { showSettings(); true }
     else -> false
}
```

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt]: no need to define a function inside a function, especially if you use it just once

https://www.youtube.com/watch?v=ihdEjDoXOgc

Just define `private fun LoadScheduleResult.hasFavorites() = !schedule.isEmpty` as a method of the FavoritesPageView class

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt]: empty line

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt]: empty line

File: app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt]: why are you defining `return` here when before you left it inside the block?

File: app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt]: when there are three or more cases it's better to use `when`

File: app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt]: this should go into a companion object

File: app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt]: you can inline this with the declaration

File: app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt]: you can use Kotlin's `fold` here

File: app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt]: empty line

File: app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt]: empty line

File: app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt]: empty line

File: app/src/main/java/net/squanchy/favorites/view/FavoritesListView.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesListView.kt]: layout here too

File: app/src/main/java/net/squanchy/favorites/view/FavoritesListView.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesListView.kt]: inline with the declaration and define as `val`

File: app/src/main/java/net/squanchy/favorites/view/FavoritesListView.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesListView.kt]: empty line

File: app/src/main/java/net/squanchy/favorites/view/FavoritesListView.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesListView.kt]: empty line

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt:15]: layout

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt]: `onFinishInflate()` requires api 25? 🤔

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt:28]: nice job returning `Nothing` here :)

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt]: empty line

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt]: annotate with `@DrawableRes`

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutPresenter.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutPresenter.kt:9]: I'm not really sure about this. I'm not a fan of having any Android dependency at all in a presenter. Let's wait for a feedback from @rock3r on this one

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutView.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutView.kt]: empty line

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutView.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutView.kt]: `@DrawableRes`

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutPresenter.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutPresenter.kt]: empty line

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutPresenter.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutPresenter.kt]: unless you can inline in, and in this case you can't, let's use the good old
```
if (condition) { 
  block() 
} else {
  otherBlock()
}
```

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt]: empty line

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutPresenter.kt
Comment (quiro91) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutPresenter.kt]: empty line

**Review by quiro91 — CHANGES_REQUESTED** [review]
Lot of changes regarding the code styling, I'm sorry that the CI didn't pick them up but unfortunately Detekt still doesn't have everything we need as Checkstyle tool. Thanks again for your help and keep up the good work!

File: app/src/main/java/net/squanchy/favorites/FavoritesComponent.kt
Comment (vishnus1224) [reply app/src/main/java/net/squanchy/favorites/FavoritesComponent.kt:25]: Detekt warns about NoNewlineAtTheEndOfFile. In this case, the empty after "}" needs to be removed?

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt
Comment (vishnus1224) [reply app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt]: What needs to be changed here?

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt
Comment (vishnus1224) [reply app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt]: Good spot 👍 Leftover from previous code. Removed it.

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutPresenter.kt
Comment (vishnus1224) [reply app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutPresenter.kt:9]: There is no Android dependency dependency here at all. The view is just a Java interface.

Comment (quiro91) [reply app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayoutPresenter.kt:9]: Oh I see, I confused it with `FavoritesSignedInEmptyLayout`

File: app/src/main/java/net/squanchy/favorites/FavoritesComponent.kt
Comment (rock3r) [reply app/src/main/java/net/squanchy/favorites/FavoritesComponent.kt:25]: This is not the end of a file tho. This is the empty line that we should have between two blocks.

File: app/src/main/java/net/squanchy/favorites/FavoritesComponent.kt
Comment (rock3r) [inline app/src/main/java/net/squanchy/favorites/FavoritesComponent.kt]: Can you please take this out?

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt
Comment (rock3r) [inline app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt]: One per line please

File: app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt
Comment (rock3r) [inline app/src/main/java/net/squanchy/favorites/FavoritesPageView.kt]: This should be immediately after `onFinishInflate()` so the code reads top-down

File: app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt
Comment (rock3r) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt]: This could be an expression body

File: app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt
Comment (rock3r) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt]: This could be an expression body

File: app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt
Comment (rock3r) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt]: This could be an expression body

File: app/src/main/java/net/squanchy/favorites/view/FavoritesListView.kt
Comment (rock3r) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesListView.kt]: One per line thanks

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt
Comment (rock3r) [reply app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt:15]: Aka one per line thanks

File: app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt
Comment (rock3r) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesSignedInEmptyLayout.kt]: This could be an expression body

File: app/src/test/java/net/squanchy/favorites/FavoritesSignedInEmptyLayoutPresenterTest.kt
Comment (rock3r) [inline app/src/test/java/net/squanchy/favorites/FavoritesSignedInEmptyLayoutPresenterTest.kt]: Missing empty line at end of file

**Review by rock3r — COMMENTED** [review]
Nice one, just a few minor formatting issues then ready to go!

**Comment by rock3r** [comment]
@vishnus1224 could you please resolve the conflicts?

File: app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt
Comment (rock3r) [inline app/src/main/java/net/squanchy/favorites/view/FavoritesAdapter.kt]: One per line please

**Comment by vishnus1224** [comment]
@rock3r  - Sorry but I tried to rebase develop on my branch but did not find any changes. How do I know where exactly the conflicts are?

**Comment by rock3r** [comment]
@vishnus1224 according to GitHub (in the box below the comments, where the merge button would be:

image

To see conflicts and solve them manually, pull from this repository's `develop` branch and merge into you own branch.

**Comment by vishnus1224** [comment]
@rock3r - Damn.. I pulled the `develop` of my fork and nothing was showing up. I'm not entirely sure how I can link back to this `repo`

**Comment by rock3r** [comment]
That's because your own fork's `develop` is not up-to-date with this repo's `develop`.

You need to add this repository as a remote:
```bash
git remote add mainline git@github.com:squanchy-dev/squanchy-android.git
```

Then checkout your local `develop` and merge this repo's `develop` into it:

```bash
git checkout develop
git pull mainline develop
```

After that you can merge the now up-to-date local `develop` into your current branch:

```bash
git checkout #333_convert_favourites_to_kotlin
git merge develop
```

That should work just fine. If you use a visual Git tool, how to perform the same steps really depends on what you use, but the steps are essentially the same.

**Review by rock3r — APPROVED** [review]
LGTM now, thanks a lot for you help and patience @vishnus1224 💯 

All yours @quiro91

**Review by quiro91 — APPROVED** [review]

**Comment by quiro91** [comment]
nice! Ready to go!

**Pull request merged**
